### DISCUSSION TITLE
Open to clarifying Linux & MacOS ARM experimental status?

### DISCUSSION THREAD

**Discussion by Alex-Monahan** [cockroachdb/cockroach#115165] [General]
Hello! 

Thanks for the great database!

Would you be open to clarifying a bit about the risks of using the experimental ARM on Linux and ARM on MacOS binaries? 

I am ok if the performance is less than optimal, but is there any risk of crashing, corruption, or data loss? If I have a small database that is not performance sensitive, will it be safe to use these binaries?

Thanks!

**Comment by Yash-xoxo** [comment]
Great question! Here's what "experimental" typically means for CockroachDB ARM builds:

**Performance:** Not as optimized as x86_64 - some codepaths may be slower

**Stability:** Generally stable for basic workloads, but less tested in production

**Risk level:**
- **Crashes:** Possible but rare - ARM builds run the same code, just different compilation target
- **Corruption/data loss:** Very unlikely - storage layer is architecture-independent
- **Edge cases:** More likely to hit untested scenarios

**For your use case (small, non-performance-sensitive DB):**
Should be fine! Just:
- Test your specific workload first
- Keep backups (you should anyway)
- Monitor for any unusual behavior
- Consider upgrading to stable when ARM support graduates from experimental

The CockroachDB team can provide more specific guidance on current ARM status, but generally "experimental" = "works but not battle-tested at scale."

**Comment by kenliu-crl** [comment]
@Alex-Monahan The post above seems to be AI-generated. Please ignore.

All currently-supported releases of CRDB (23.2+ at the time of this writing) are supported for production use on Linux ARM.

MacOS builds on ARM (i.e. Apple Silicon) are marked as experimental, but you should have no problem running a non-production cluster on a Mac. All of our internal Engineering staff develop CRDB on Macs.
